Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -918,6 +966,146 @@ lemma nonsingular [Nontrivial R] {x y : R} (h : E.equation x y) : E.nonsingular
   E.nonsingular_of_Δ_ne_zero h <| E.coe_Δ' ▸ E.Δ'.ne_zero
 #align elliptic_curve.nonsingular EllipticCurve.nonsingular
 
+section ModelsWithJInvariant_0_1728
+
+variable (R : Type u) [CommRing R]
+
+/-- The elliptic curve $Y^2 + Y = X^3$ when $3$ is invertible. -/
+def ofJInvariant0 [Invertible (3 : R)] : EllipticCurve R :=
+  ⟨
+    WeierstrassCurve.ofJInvariant0 R,
+    ⟨-27, -(⅟3) ^ 3,
+      by
+        rw [calc (27 : R) = (3 : R) ^ 3 := by norm_num]
+        simp only [mul_neg, neg_mul, neg_neg]
+        exact pow_mul_pow_eq_one 3 (mul_invOf_self (3 : R))
+      , by
+        rw [calc (27 : R) = (3 : R) ^ 3 := by norm_num]
+        simp only [mul_neg, neg_mul, neg_neg]
+        exact pow_mul_pow_eq_one 3 (invOf_mul_self (3 : R))
+    ⟩,
+    by rw [WeierstrassCurve.ofJInvariant0_Δ R]
+  ⟩
+
+lemma ofJInvariant0_j [Invertible (3 : R)] : (ofJInvariant0 R).j = 0 := by
+  simp only [j, ofJInvariant0, WeierstrassCurve.ofJInvariant0_c₄]
```

Comment:
A lot of these can probably be `rw`'s. I also find it strange why `ring1` works here but not `norm_num1`.